borg/archiver.py

```diff
@@ -406,64 +406,127 @@ def item_is_hardlink_master(item):
+            # regular file is replaced with a link, or a symlink changes to a
+            # hard link, it is indicated in compare_mode instead.
+            if b'source' not in item1:
+                if item1.get(b'deleted'):
+                    return 'added {}'.format(describe_link(item2))
+            elif b'source' not in item2:
+                if item2.get(b'deleted'):
+                    return 'removed {}'.format(describe_link(item1))
+            elif (item1[b'source'] != item2[b'source'] and
+                  get_mode(item1)[0] == get_mode(item2)[0]):
+                return 'link changed'
+
+        def compare_content(item1, item2):
+            if (not can_compare_chunk_ids or
+                    item1[b'chunks'] != item2[b'chunks']):
+                chunk_ids1 = [c[0] for c in item1[b'chunks']]
+                chunk_ids2 = [c[0] for c in item2[b'chunks']]
+                chunk_id_set1 = set(chunk_ids1)
+                chunk_id_set2 = set(chunk_ids2)
+                total1 = sum_chunk_size(item1)
+                total2 = sum_chunk_size(item2)
+
+                if (can_compare_chunk_ids or total1 != total2 or
+                    not fetch_and_compare_chunks(chunk_ids1, chunk_ids2,
+                                                    archive1, archive2)):
+                    added = sum(c[1] for c in (chunk_id_set2 - chunk_id_set1))
+                    removed = -sum(c[1] for c in (chunk_id_set1 - chunk_id_set2))
+
+                    if total1 is None:
+                        return 'added {:>11}'.format(format_file_size(total2))
+                    elif total2 is None:
+                        return 'removed {:>9}'.format(format_file_size(total1))
+                    else:
+                        return '{:>8} {:>8}'.format(format_bytes(added),
+                                                    format_bytes(removed))
+
+        def compare_directory(item1, item2):
+            if not item1.get(b'deleted') and item2.get(b'deleted'):
+                if get_mode(item1)[0] == 'd':
+                    return 'removed directory'
+            elif not item2.get(b'deleted') and item1.get(b'deleted'):
+                if get_mode(item2)[0] == 'd':
+                    return 'added directory'
+
+        def compare_owner(item1, item2):
+            user1, group1 = get_owner(item1)
+            user2, group2 = get_owner(item2)
+            if user1 != user2 or group1 != group2:
+                return '[{}:{} -> {}:{}]'.format(user1, group1, user2, group2)
+
+        def compare_mode(item1, item2):
+            mode1 = get_mode(item1)
+            mode2 = get_mode(item2)
+
+            if mode1 != mode2:
+                return '[{} -> {}]'.format(mode1, mode2)
+
         def compare_items(path, item1, item2, deleted=False):
             """
             Compare two items with identical paths.
             :param deleted: Whether one of the items has been deleted
             """
+            line = []
```

Comment:
instead of "line", you could use the more meaningful "changes". plural also indicates better that this is a list.